### ISSUE TITLE
Incorrect highlighting of type annotated monadic bound variable

### ISSUE THREAD

**Issue by andrewthad** [neovimhaskell/haskell-vim#67]
If you have a program that looks like this:

    myFunc :: IO ()
    myFunc = do
      x :: Int <- readLn
      print x

Then the first occurrence of `x` ends up being highlighted like it's a function declaration.

**Comment by raichoo** [comment]
Yep, that is known. I don't see a way to fix this other than putting parentheses around this like so:

`(x :: Int) <- readLn`

VimL lacks the ability to highlight based on context.
